index.d.ts

```diff
@@ -1161,7 +1161,19 @@ declare namespace plugins {
      * the key/value pairs to record. For example, using
      * `variables => variables` would record all variables.
      */
-    variables?: string[] | ((variables: { [key: string]: any }) => { [key: string]: any });
+    variables?:
+      | string[]
+      | ((
+          variables: { [key: string]: any },
+          info: {
+            span?: Span;
+            args?: ExecutionArgs;
```

Comment:
Can this be made more granular? Passing this entire object directly might end up passing a structure that may change over time. Normalizing only what is needed (like what you did with `operation`) makes it easier to ensure the structure won't change.

Reply:
Hmmm, in theory yes but I'm wondering if that's less of a concern here given args is already exposed for the execute hook? 

```
execute?: (span?: Span, args?: ExecutionArgs, res?: any) => void;
```

I'd personally prefer to keep the same types when they're shared so that there's less mental overhead of wondering what's in either.